melody: init at 2.2.1

hotlineguy opened the pull request:
###### Motivation for this change
Add the Melody Package (https://github.com/artemanufrij/playmymusic) to the package repository.

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

First, well second, time commiting. Apparently, i managed to mess up the former branch. I'm sorry about that.

hotlineguy committed:
melody: init at 2.2.1
Add the Melody Package (https://github.com/artemanufrij/playmymusic) to the package repository.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

First, well second, time commiting. Apparently i managed to mess up the former branch.
I'm sorry about that.

drewrisinger commented on pkgs/applications/audio/melody/default.nix:
```diff
@@ -0,0 +1,74 @@
+{ stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, gtk3
+, pkgconfig
+, python3
+, desktop-file-utils
+, vala
+, wrapGAppsHook
+, appstream
+, clutter-gtk
+, cmake
+, gst_all_1
+, json-glib
+, libgee
+, libsoup
+, pantheon
+, sqlite
+, taglib
```
Sort these alphabetically

drewrisinger commented on pkgs/applications/audio/melody/default.nix:
```diff
@@ -0,0 +1,74 @@
+  src = fetchFromGitHub {
+    owner = "artemanufrij";
+    repo = "playmymusic";
+    rev = "bba5fe74134ad7533c41044d8ce41f1a8f69b3b0";
```
Use a tagged release if possible
```suggestion
    rev = "2.2.1";
```
If not using a tagged release, a reason should be given, and the version above should be changed to something like ``version = "unstable-2020-09-14"`` (where the date is the date of the commit).

drewrisinger commented on pkgs/applications/audio/melody/default.nix:
```diff
@@ -0,0 +1,74 @@
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils
+    meson
+    ninja
+    pkgconfig
+    python3
+    vala
+    wrapGAppsHook
```
Sort alphabetically

drewrisinger commented on pkgs/applications/audio/melody/default.nix:
```diff
@@ -0,0 +1,74 @@
+    wrapGAppsHook
+  ];
+
+  buildInputs = with gst_all_1; [
+    appstream
+    clutter-gtk
+    cmake
+    gst-plugins-bad
+    gst-plugins-base
+    gst-plugins-good
+    gst-plugins-ugly
+    gstreamer
+    gtk3
+    json-glib
+    libgee
+    libsoup
+    pantheon.elementary-icon-theme
+    pantheon.granite
+    sqlite
+    taglib
+  ];
```
To clarify which are from ``gst_all_1``, I recommend splitting this into two lists:

```
buildInputs = [
  non-gst_all_1_pkgs
] ++ 
(with gst_all_1; [ gst_all_1_pkgs ]);

drewrisinger commented on pkgs/applications/audio/melody/default.nix:
```diff
@@ -0,0 +1,74 @@
+
+  postPatch = ''
+    chmod +x meson/post_install.py
+    patchShebangs meson/post_install.py
```
Is this script automatically run?

drewrisinger commented on pkgs/applications/audio/melody/default.nix:
```diff
@@ -0,0 +1,74 @@
+  meta = with stdenv.lib; {
+    description = "A music player for listening to local music files, online radio and Audio CD's - Designed for elementary OS";
+    homepage = "https://anufrij.org/melody";
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Plus;
```
Using ``gpl3`` is deprecated. The source seems to hint that GPL3 or later are acceptable licenses.

drewrisinger commented on pkgs/top-level/all-packages.nix:
```diff
@@ -27968,4 +27968,6 @@ in
 
   unifi-poller = callPackage ../servers/monitoring/unifi-poller {};
 
+  melody = callPackage ../applications/audio/melody {};
+
```
This typically wouldn't get put at the end, but would probably be sorted into semi-alphabetical order in the range of lines 20000+ (don't have the file in front of me, so rough range)

drewrisinger reviewed (changes requested):
Hello, and thanks for contributing your first package to ``nixpkgs``!

I have several comments, mostly formatting & style.

1. Separate out adding yourself as a maintainer to a separate commit. ``git reset --soft HEAD~1``, then commit them separately.
2. Several comments in code body.

My standard review checklist:
* [ ] Diff needs some improvement
* [ ] Maintainer commit needs separated
* [x] Builds via nix-review:

```
https://github.com/NixOS/nixpkgs/pull/97995
1 package built:
melody
```

hotlineguy committed:
maintainers: add hotlineguy

Add myself to maintainers-list.nix

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

hotlineguy committed:
maintainers: add hotlineguy
add myself to maintainers-list.nix

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

hotlineguy committed:
Merge branch 'melody' of https://github.com/hotlineguy/nixpkgs into melody

hotlineguy commented:
Apparently, i'm able to mess up things but not to fix em. Closing this and might pick up up later on...
